.github/workflows/_charm-release.yaml

```diff
@@ -136,27 +142,22 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
-      - name: Select charmhub channel
-        uses: canonical/charming-actions/channel@2.1.1
-        id: channel
       - name: Fetch charm artifacts
         uses: actions/download-artifact@v3
         with:
           name: charms
           path: "${{ github.workspace }}/${{ inputs.charm-path }}"
-      - name: Upload charm to charmhub
-        uses: canonical/charming-actions/upload-charm@2.5.0-rc
-        with:
-          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
-          github-token: "${{ secrets.GITHUB_TOKEN }}"
-          channel: "${{ steps.channel.outputs.name }}"
-          built-charm-path: "${{ matrix.path }}"
-          charm-path: "${{ inputs.charm-path }}"
-          tag-prefix: "${{ inputs.release-tag-prefix }}"
-          charmcraft-channel: "${{ inputs.charmcraft-channel }}"
-          # We set destructive mode to false, otherwise runner's OS would have to match
-          # charm's 'build-on' OS.
-          destructive-mode: false
+      - name: Upload charm to Charmhub
+        env:
+          CHARMHUB_TOKEN: ${{ secrets.CHARMHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          pip install git+https://github.com/lucabello/noctua
+          cd "${{ inputs.charm-path }}"
+          release=$(noctua charm release --path="${{ matrix.path }}" --channel=${{ inputs.release-channel }} --json | tail -n1)
+          revision=$(echo "$release" | jq .revision)
+          # TODO: push git tag; is it necessary?
+          gh release create "${{inputs.release-tag-prefix}}rev${revision}" --title="Revision ${revision}" --generate-notes
```

Comment:
A problem with adding that to `noctua` is that the tool starts doing too much; one of the complaint I had about charming-actions is that you had no idea a GitHub release was happening.

I'm trying to follow the "path of least surprises", trying to stay closer to `charmcraft` than to `charming-actions`

I'm not sure if I understand what's the benefit of making a composite action for this, considering it's one line of Bash; it would, as you said, hide the logic, and I don't see a difference in readability between an action saying ("create charm release on github") and a comment just above that line with the same message.

Reply:
I think this goes in the same direction as [this thread](https://github.com/canonical/observability/pull/207#discussion_r1750553519) so we can probably discuss it there.  Its subjective, but I like the cleanliness like you have in the [public workflows](https://github.com/canonical/observability/blob/main/.github/workflows/bundle-pull-request.yaml) - they're very easy for someone else to read and understand the intent.  As soon as run commands get involved, the clarity for a reader *who doesn't want the details* goes down